hstream/test/HStream/SpecUtils.hs

```diff
@@ -364,3 +365,21 @@ runDropSql :: HStreamClientApi -> T.Text -> Expectation
 runDropSql api sql = do
   DropPlan checkIfExists dropObj <- streamCodegen sql
   dropAction checkIfExists dropObj api `grpcShouldReturn` Empty
+
+newtype TimedOut = TimedOut UTCTime deriving (Eq, Show)
+instance Exception TimedOut
+
+testTimeout :: Int -> IO a -> IO () -> IO a
+testTimeout usec action cancel = do
+  expired <- fmap TimedOut getCurrentTime
+  pid <- myThreadId
+  parent pid expired
+  where
+    child pid expired = do
+      threadDelay usec
+      cancel
+      threadDelay 10000000
+      throwTo pid expired
+
+    parent pid expired = do
+      bracket (forkIO $ child pid expired) killThread (const action)
```

Comment:
Does [System.Timeout.timeout](https://hackage.haskell.org/package/base-4.16.0.0/docs/System-Timeout.html#v:timeout) in base package meets your requirements?

Reply:
May not, the problem is: when timeout, I need to send a response to close grpc connect first(`line 380-381` in pr), otherwise the whole test frame will hang forever even after all the tests are finished.

Reply:
fixed